vignettes/analyzing-load-test-logs.Rmd

```diff
@@ -12,31 +12,268 @@ vignette: >
+
+Once the JavaScript files have finished loading, communication to a data server can begin.  For Shiny applications, this involves communicating over a [WebSocket](https://en.wikipedia.org/wiki/WebSocket).  A WebSocket is a single, constant line of communication between the Shiny server and the Shiny application in the browser.  The WebSocket will notify the Shiny server about which input values have changed by a user and will also notify the Shiny application about which outputs need to been updated.  This communication process will continue until the WebSocket is closed, typically the last line of the `recording.log` file.
+
+As an example, the basic [`01_hello` gallery shiny app](https://gallery.shinyapps.io/001-hello/) loads the homepage, loads more than 20 supporting CSS and JavaScript files, and then begins WebSocket communication with the supporting Shiny server.
+
+
+# Report Output
+
+The data provided by `shinyloadtest::tidy_loadtest` can be used to directly create a stand alone HTML report.  This report contains six major sections of diagnostic information.
+
+```{r, include = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  `1 user` = "../test_sessions/demo1",
+  `4 users` = "../test_sessions/demo4",
+  `16 users` = "../test_sessions/demo16"
+)
+# ignore <- shinyloadtest::make_report(df, "report.html", verbose = FALSE, open_browser = FALSE)
+```
+```{r, eval = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  "1 user" = "./demo1",
+  "4 users" = "./demo4",
+  "16 users" = "./demo16"
+)
+shinyloadtest::make_report(df, "report.html")
+```
+
+## Sessions
+
+The `Sessions` tab of the shinyloadtest report displays the each simulated session performed by each user.  Each run displays their respective [gantt chart](https://en.wikipedia.org/wiki/Gantt_chart) by clicking on the named buttons near the top of the webpage.
+
+```{r, echo = FALSE, fig.height = 3}
+suppressPackageStartupMessages(library(dplyr, quiet = TRUE))
+suppressPackageStartupMessages(library(shinyloadtest, quiet = TRUE))
+df %>%
+  filter(run == "4 users") %>%
+  plot_gantt()
+```
+
+Sessions that are desaturated in color are sessions that have started before (warmup) or ended after (cooldown) the maintenance period.  The maintenance period is marked with vertical dotted lines spanning every simulated user.
+
+In the example above, four users were stagger started and simulated for 10 minutes. The majority of the time is spent calculating new values for the outputs in the recording.  The session gantt chart below displays the same recording but with sixteen users.  
+
+```{r, echo = FALSE, fig.height = 5}
+df %>%
+  filter(run == "16 users") %>%
+  plot_gantt()
+```
+
+Both runs were executed on a single processor.  With very high resource contention, a larger proportion of time is spent calculating in the `16 users` run.
```

Comment:
This sentence is confusing because it is not clear if you are referring to shinycannon or the app. I'd say, "both runs were against a Shiny application supported by a single R process".